doc/classes/Object.xml

```diff
@@ -38,78 +42,119 @@
 			<return type="Variant" />
 			<param index="0" name="property" type="StringName" />
 			<description>
-				Virtual method which can be overridden to customize the return value of [method get].
-				Returns the given property. Returns [code]null[/code] if the [param property] does not exist.
+				Override this method to customize the behavior of [method get]. Should return the given [param property]'s value, or [code]null[/code] if the [param property] does not exist.
 			</description>
 		</method>
 		<method name="_get_property_list" qualifiers="virtual">
 			<return type="Dictionary[]" />
 			<description>
-				Virtual method which can be overridden to customize the return value of [method get_property_list].
-				Returns the object's property list as an [Array] of dictionaries.
-				Each property's [Dictionary] must contain at least [code]name: String[/code] and [code]type: int[/code] (see [enum Variant.Type]) entries. Optionally, it can also include [code]hint: int[/code] (see [enum PropertyHint]), [code]hint_string: String[/code], and [code]usage: int[/code] (see [enum PropertyUsageFlags]).
+				Override this method to customize how script properties should be handled by the engine.
+				Should return a property list, as an [Array] of dictionaries. The result is added to the array of [method get_property_list], and should be formatted in the same way. Each [Dictionary] must at least contain the [code]name[/code] and [code]type[/code] entries.
+				The example below displays [code]hammer_type[/code] in the Inspector dock, and serializes it, only if [code]using_hammer[/code] is [code]true[/code]:
+				[codeblock]
+				@tool
+				extends Node2D
+
+				@export var using_hammer = false:
+				    set(value):
+				        using_hammer = value
+				        notify_property_list_changed()
+				var hammer_type = 0
+
+				func _get_property_list():
+				    var properties = []
+
+				    if using_hammer:
+				        properties.append({
+				            "name": "hammer_type",
+				            "type": TYPE_INT,
+				            "usage": PROPERTY_USAGE_DEFAULT,
+				            "hint": PROPERTY_HINT_ENUM,
+				            "hint_string": "Wooden,Iron,Golden,Enchanted"
+				        })
+
+				    return properties
+				[/codeblock]
```

Comment:
This example is not complete, because it's missing `_set()` and `_get()`. The property will just appear in the inspector without being saved. See https://github.com/godotengine/godot/issues/68797#issuecomment-1321258302

Reply:
Feels like an oversight, I don't recall this happening in 3.x...

Reply:
It was always like that. The properties you return in `_get_property_list()` don't even have to exist.

Reply:
They do not have to exist, but it was under my assumption that it would save.

In fact, I just tested the snippet of code, and the `hammer_type` property does save completely fine, at least within the currently edited scene.

Reply:
It seems to be always 0 though.
I think the method should at least mention `_get()` and `_set()`, because they are closely related. (and these 2 methods should have some examples)

Reply:
I was trying to provide examples for `_get()` and `_set()`, and they would not work as I expected them to (nor as how they worked in 3.x), and I can't tell if it's me missing something or an oversight.

Reply:
In the issue I linked there is a working example.

Reply:
As mentioned in the linked issue, you don't need `_get` and `_set` if your `_get_property_list` defines a property with the same name as an existing class member. You only need those two methods for purely computed properties.

See https://github.com/Arnklit/ShellFurGodot/blob/main/addons/shell_fur/shell_fur_manager.gd as an example of this being used in a 3.x plugin from a couple of years ago.

So this code snippet should be valid in principle, though I've not tested it to see if there are some other issues.